cmd/ebrelayer/txs/registry.go

```diff
@@ -13,44 +13,51 @@ import (
 	bridgeregistry "github.com/Sifchain/sifnode/cmd/ebrelayer/contract/generated/artifacts/contracts/BridgeRegistry.sol"
 )
 
-// TODO: Update BridgeRegistry contract so that all bridge contract addresses can be queried
-//		in one transaction. Then refactor ContractRegistry to a map and store it under new
-//		Relayer struct.
-
 // ContractRegistry is an enum for the bridge contract types
 type ContractRegistry byte
 
 const (
-	// Valset valset contract
-	Valset ContractRegistry = iota + 1
-	// Oracle oracle contract
-	Oracle
 	// BridgeBank bridgeBank contract
-	BridgeBank
+	BridgeBank ContractRegistry = iota + 1
 	// CosmosBridge cosmosBridge contract
 	CosmosBridge
 )
 
+var bridgeBankAddress = common.HexToAddress(nullAddress)
+var cosmosBridgeAddress = common.HexToAddress(nullAddress)
+
 // String returns the event type as a string
 func (d ContractRegistry) String() string {
-	return [...]string{"valset", "oracle", "bridgebank", "cosmosbridge"}[d-1]
+	return [...]string{"bridgebank", "cosmosbridge"}[d-1]
 }
 
 // GetAddressFromBridgeRegistry queries the requested contract address from the BridgeRegistry contract
 func GetAddressFromBridgeRegistry(client *ethclient.Client, registry common.Address, target ContractRegistry,
 	sugaredLogger *zap.SugaredLogger) (common.Address, error) {
+	// Return address if already got and stored
+	switch target {
+	case BridgeBank:
+		if bridgeBankAddress.Hex() != nullAddress {
+			return bridgeBankAddress, nil
+		}
+	case CosmosBridge:
+		if cosmosBridgeAddress.Hex() != nullAddress {
+			return cosmosBridgeAddress, nil
+		}
+	default:
+		panic("invalid target contract address")
```

Comment:
Why do we panic instead of throwing error here?

Reply:
The relayer should never go to this branch, and panic can make relayer exit, then we can fix the issue. If return error, relayer can do nothing, and this error not depends on any runtime data, so it will be definitely failed again. the relayer can't work normally anymore.